### PR TITLE
Removed toJS calls on getMarkers and unnecessary find calls

### LINKED ISSUE #3466: Remove toJS and refactor find calls on getMarkers function
Make the getMarkers function faster, probably there is other reasons why is slow but these are the obvious ones.
1. There is abusive and repeated usage of .toJS inside
2. here is repeated calls to .find also

The multiple calls seems to be because of not needed re-renders, ideally the function gets called only once.

A subtask for #3444

### PR DESCRIPTION
**Pull request by grafitto**
fixes #3466 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/react/Library/components/MapView.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class MapView extends Component {
             <b>{markers.get('totalRows')}</b> {t('System', 'documents')}
           </span>
         </div>
-        <Markers entities={markers.get('rows')}>
+        <Markers entities={markers.get('rows')} storeKey={storeKey}>
```
Comment (konzz) [inline app/react/Library/components/MapView.js:72]: Why does the Markers component need the storeKey property?

File: app/react/Map/helper.js [hunk 2]
```diff
@@ -41,105 +57,106 @@ const markersToStyleFormat = markers =>
...
-          .forEach(({ value: point }) => {
-            const { lat, lon } = point;
-            markers.push({
-              properties: { entity, color, info: point.label },
-              latitude: lat,
-              longitude: lon,
-              label,
+  const geolocationPropNames = geolocationProps.map(prop => prop.get('name'));
+
+  if (geolocationProps.size && metadata) {
```
Comment (konzz) [inline app/react/Map/helper.js:64]: Please refactor the nested if and else into functions

File: app/react/Map/specs/__snapshots__/helper.spec.js.snap [hunk 3]
```diff
@@ -3,11 +3,12 @@
 exports[`map helper getMarkers() Inheritance should include inherited properties 1`] = `
```
Comment (konzz) [inline app/react/Map/specs/__snapshots__/helper.spec.js.snap:3]: Change the spec that is using this snapshot to test that the inherited property is in the array of markers, an snapshot here is confusing and hard to tell when the snapshot is correct

**Review by konzz — CHANGES_REQUESTED** [review]

File: app/react/Map/helper.js [hunk 4]
```diff
@@ -1,5 +1,21 @@
+/* eslint-disable max-statements */
 /** @format */
```
[the remark this answers is not in the document]
Comment (grafitto) [reply app/react/Map/helper.js:2]: Okay

[the remark this answers is not in the document]
Comment (grafitto) [reply app/react/Map/helper.js]: Removed

File: app/react/Library/components/MapView.js (hunk 1 above)
Comment (grafitto) [reply app/react/Library/components/MapView.js:72]: It actually does not need storeKey, I forgot to remove when I realized entities are passed as a property. Removing...

Comment (grafitto) [reply app/react/Library/components/MapView.js]: Removed

File: app/react/Map/helper.js (hunk 2 above)
Comment (grafitto) [reply app/react/Map/helper.js:104]: Done

File: app/react/Map/specs/__snapshots__/helper.spec.js.snap (hunk 3 above)
Comment (grafitto) [reply app/react/Map/specs/__snapshots__/helper.spec.js.snap:3]: Done, kindly check if that is what is needed.

**Pull request merged**